File: Simulation/14719번 - 빗물/14719.cpp
#include <iostream>

using namespace std;

int h, w;
int block[501];

void input() {
    cin >> h >> w;
    for(int i = 0; i < w; i++) {
        cin >> block[i];
    }
}

void solve() {
    int ans = 0;
    
    for(int i = 0; i < w; i++) {
        int left = 0, right = 0;
        
        for(int j = 0; j < i; j++) left = max(left, block[j]);
        for(int j = i + 1; j < w; j++) right = max(right, block[j]);
        if(left == 0 || right == 0) continue;
        
        int rain = min(left, right);
        if(rain > block[i]) ans += rain - block[i];
    }
    
    cout << ans << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Simulation/14719번 - 빗물/14719.py
import sys

input = sys.stdin.readline

h, w = map(int, input().split())
water = list(map(int, input().split()))

total = 0

for i in range(w):
    left = 0
    right = 0
    for _ in range(0, i):
        left = max(left, water[_])
    for _ in range(i + 1, w):
        right = max(right, water[_])
    if left != 0 and right != 0 and (min(left, right) > water[i]):
        cnt = min(left, right) - water[i]
        total += cnt
        
print(total)